task/mongo-optimization-task.js

```diff
@@ -241,22 +261,21 @@ async function task_3_1(db) {
             }
         },
         {
-            "$lookup" : {
-                "from" : "clientCriteria",
-                "localField" : "criteria_value",
-                "foreignField" : "value",
-                "as" : "criteria"
-            }
-        },
-        {
-            "$unwind" : "$criteria"
-        },
-        {
-            "$unwind" : "$criteria.versions"
-        },
-        {
-            "$match" : {
-                "criteria.versions.initiativeId" : ObjectId("58af4da0b310d92314627290")
+            "$lookup": {
+                "from": "clientCriteria",
+                let: { "value": "$criteria_value" },
+                pipeline: [
+                    {
+                        $match:
+                        {
+                            "versions.initiativeId": ObjectId("58af4da0b310d92314627290")
+                        }
+                    },
+                    {
+                        $match: { $expr: { $eq: ["$value", "$$value"] } }
+                    }
```

Comment:
you can set $limit 1 (to avoid redundant search after the first mach) and add projection.